### ISSUE TITLE
service_check_timeout is not honored

### ISSUE THREAD

**Issue by sinkingsun2** [naemon/naemon-core#261]
I used naemon + gearmand to create a distributed monitoring. 
but as I observed, sometimes, naemon think the service checking script is still running, so it won't kill it and just keep delay the next_check_time. 
![naemon](https://user-images.githubusercontent.com/22874824/45841892-005b4100-bcd0-11e8-8246-362ed321b218.jpg)
my service is actually scheduled to run every minute. 
but it's been stuck there for 2 hours.

**Comment by sinkingsun2** [comment]
if (temp_service->is_executing == FALSE)
				continue;

			/* determine the time at which the check results should have come in (allow 10 minutes slack time) */
			expected_time = (time_t)(temp_service->next_check + temp_service->latency + service_check_timeout + check_reaper_interval + 600);

**Comment by sinkingsun2** [comment]
in my case, I think if I comment out these code in checks_service.c, it will be better 
			/* skip services that are currently executing (problems here will be caught by orphaned service check) */
			if (temp_service->is_executing == TRUE)
				continue;

**Comment by sinkingsun2** [comment]
just to confirm that after I comment out these 2 lines, it worked better.

**Comment by sni** [comment]
Might be more a mod-gearman issue then. Which gearmand / libgearman is that? The main question here is why mod-gearman never sent a result for this service.

**Comment by sinkingsun2** [comment]
yes, I agree, but even if mod-gearman didn't give back any result,
naemon service should time out. right?
I used 2 gearmand on different port to improve performance, not sure if
that caused any bug in the whole architecture.
I at least can see 1 bug clearly. when I run gearman_top -H localhost:4730,
I can't see any working task in queue. but when I run gearman_top -H
localhost:4731, everything looks normal.


Sven Nierlein <notifications@github.com> 于2018年9月28日周五 上午6:49写道：

**Comment by sni** [comment]
when using mod-gearman, naemon cannot kill or timeout the service check. Thats usually a indication on a misconfiguration.
